apps/freeablo/faworld/findpath.cpp

```diff
@@ -4,11 +4,22 @@
 
 #include <misc/stdhashes.h>
 
+#include <cmath>
 #include <queue>
 
+namespace
+{
+    double distanceCost(const Misc::Point& a, const Misc::Point& b)
+    {
+        static const double diagonal_distance = sqrt(2.);
+
+        return (a.x != b.x && a.y != b.y) ? diagonal_distance : 1.;
+    }
+}
+
 namespace FAWorld
 {
-    template <typename T, typename Number = size_t> struct PriorityQueue
+    template <typename T, typename Number = double> struct PriorityQueue
```

Comment:
I don't want to use floating point calculations anywhere in the game simulation code (it's fine in rendering). This is to support the lock-step multiplayer implemnentation: https://github.com/wheybags/freeablo/blob/master/decisions.md#mon-apr-30-084155-cest-2018m

If you need to, you can use the FixedPoint class, or better still just avoid fractional units entirely.

Reply:
In this case, you could just return 3 for diagonal for  diagonal, and 2 for straight, which would approximate the ratios for sqrt(2) and 1. Should probably leave a comment explaining why too.

Reply:
Agreed. "double" should be replaced by 3/2 or 14/10 (which is more close).